### ISSUE TITLE
Better support for GNOME 40

### ISSUE THREAD

**Issue by spacekookie** [mzur/gnome-shell-wsmatrix#188]
Hey!

I'm not a gnome plugin developer, so I don't know if this is technically possible.
Previously (i.e. on Gnome 3.18) the workspaces were "physically" laid out like a grid, meaning that when jumping a diagonal, the camera would move diagonally.

Starting in Gnome 40 the workspaces are now laid out in a line, meaning that when jumping a diagonal in the grid a workspace is skipped horizontally. This is giving me quite a bit of vertigo and not very intuitive to use.

Is this something that can be fixed?
Thank you for maintaining this extension nonetheless. It feels like Gnome vanilla gets less and less usable as the years go on...

**Comment by mzur** [comment]
You mean the animation in the activity overview? This is tracked in #183. I've tinkered with a 2D layout there a while back and @ebeem added [preliminary support](https://github.com/mzur/gnome-shell-wsmatrix/blob/03fe291efa4a855e3113165d9aa4851093dca49b/wsmatrix%40martin.zurowietz.de/overview/workspacesView.js#L97-L102) for that in his initial PR for GNOME 40. However, this is a tricky beast and I was happy that we managed to finish initial GNOME 40 support at all. It can take a while before we start to tackle the remaining issues with GNOME 40 (probably until I start using it myself :smile:).

**Comment by ebeem** [comment]
> aware though that when switching workspaces this way the popup showing you what workspace you're on doesn't render for very long? (it only pops up on screen for ~500ms)

@spacekookie 

you might want to go to the extension preferences and alter this behavior with the option `time to show the popup (ms)`, you can control the time from there or set it to `0` so the popup disappears upon releasing the modifier keys `Ctrl+Alt`.

Please let us know if you have any issues that we need to address here other than #183 so we can keep the list of issues clean.
Thanks for your feedback!

**Comment by bradleesand** [comment]
What I've been seeing since I upgraded to GNOME 40 is that, when I first boot up, the extension appears to be enabled but not actually arranging the workspaces in a grid. Then, if I toggle the extension off and back on again, it starts working correctly and arranges the workspaces properly in a grid. Perhaps the extension is crashing? I don't know where to find logs for that.

I've also been experiencing issues where something like X or GNOME seems to crash and restart: screen goes black for a moment and when it starts working again, all the windows that were on my second screen have moved over to the first screen. This mostly seems to be correlated with Zoom and Guake, not necessarily this extension; however, when things start working again, this extension seems to have crashed or something because I have to toggle it off and on again to get the grid to work.

**Comment by ebeem** [comment]
@bradleesand 
you can try out `journalctl -f /usr/bin/gnome-shell` to check the logs and report the warning/error messages you get.
I don't think that we received such an issue you before, so it will be a good idea to get the logs and post it so we can investigate more.

**Comment by mzur** [comment]
Please comment or reopen if you have something to add that is not already discussed in another issue.
